### ISSUE TITLE
Timeouts

### ISSUE THREAD

**Issue by Boimb** [AlCalzone/node-tradfri-client#110]
Hi,
Now, that i'm OK with my dev on MacOS, I pushed it on a Raspberry Pi.
Can discover the gateaway, but get timeout when I try to authenticate.
Is someone aware of something like this ? I can't imagine I'm the only one to use that lib on Pi.
(same issue on win 7)
node-tradfri-client: 1.2.1 and 1.3.4 tested
node: 8.9.4

Thx for your help.
Boimb.

**Comment by AlCalzone** [comment]
If you are using the ip to connect, please try the host name and vice versa.

**Comment by Boimb** [comment]
I tried all 3 ways to connect: host, IPV4 & IPV6
And always the same error
```
0|gladys   | { Error: The gateway did not respond in time.
0|gladys   |     at TradfriClient.<anonymous> (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:172:39)
0|gladys   |     at Generator.next (<anonymous>)
0|gladys   |     at fulfilled (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:4:58)
0|gladys   |     at <anonymous>
0|gladys   |     at process._tickDomainCallback (internal/process/next_tick.js:228:7) code: 1 }

0|gladys   | Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
0|gladys   |     at Socket._healthCheck (dgram.js:637:11)
0|gladys   |     at Socket.send (dgram.js:435:8)
0|gladys   |     at processInterface (/home/pi/gladys/api/hooks/tradfri/node_modules/mdns-server/index.js:335:37)
0|gladys   |     at SendWrap.callback (/home/pi/gladys/api/hooks/tradfri/node_modules/mdns-server/index.js:342:15)
0|gladys   |     at SendWrap.afterSend [as oncomplete] (dgram.js:503:8)
```
Gladys refers to the app in which node-tradfri-client is executed. (Works well on my Gladys dev install on Mac)

It seems that the "code 1" reference at line 6 doesn't show each time.

**Comment by AlCalzone** [comment]
Can you please show the code you are using to connect? I’m using the lib in an iobroker adapter which is mainly used on raspberry and without any problems.

**Comment by Boimb** [comment]
Hi,
Here's the pretty simple code that fails.
````
// test.js
const {TradfriClient, discoverGateway} = require ('node-tradfri-client')
discoverGateway()
  .then(gateway => {
    console.log('gateway: ', gateway)
    let client = new TradfriClient(gateway.host); // create client
    console.log('client: ', client)
    const secret = '#############'
    return client.authenticate(secret)
  })
  .then(({identity, psk}) => {
    console.log('identity: ', identity)
    console.log('PSK: ', psk)
    return {}
})
  .catch(console.log)
````
Then
````
gateway:  { name: 'gw-b8d7af2bba0f',
  host: 'TRADFRI-Gateway-b8d7af2bba0f.local',
  version: '1.2.42',
  addresses: [ '169.254.10.225', 'fe80::bad7:afff:fe2b:ba0f' ] }
client:  TradfriClient {
  domain: null,
  _events: {},
  _eventsCount: 0,
  _maxListeners: undefined,
  hostname: undefined,
  observedPaths: [],
  devices: {},
  groups: {},
  ipsoOptions: {},
  rememberedObserveCallbacks: Map {},
  requestBase: 'coaps://undefined:5684/' }
TypeError: Cannot read property 'startsWith' of undefined
    at normalizeHostname (/home/pi/gladys/api/hooks/tradfri/node_modules/node-coap-client/build/CoapClient.js:108:19)
    at Function.setSecurityParams (/home/pi/gladys/api/hooks/tradfri/node_modules/node-coap-client/build/CoapClient.js:121:20)
    at TradfriClient.<anonymous> (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:145:43)
    at Generator.next (<anonymous>)
    at /home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:3:12)
    at TradfriClient.tryToConnect (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:142:16)
    at TradfriClient.<anonymous> (/home/pi/gladys/api/hooks/tradfri/node_modules/node-tradfri-client/build/tradfri-client.js:169:32)
````
Same goes when tryin 
````
let client = new TradfriClient(gateway.addresses[0]); // create client
// or
let client = new TradfriClient(gateway.addresses[1]); // create client
````

**Comment by AlCalzone** [comment]
The error leads me to believe that your nodejs version is too old. Nonetheless you might need to use the discovered `name` (gw-...) instead of `host`.

**Comment by AlCalzone** [comment]
I’m only on mobile for the next two weeks, so I cannot check easily. For some reason undefined is passed as the host name internally.

**Comment by Boimb** [comment]
IPV4 is in ´169.254.xxx.xxx’ range. 
Seems I got a DNS issue on my local network , isn’t it ?

**Comment by AlCalzone** [comment]
Might be but I wouldn’t rule out another problem. `coaps://undefined` looks like something is wrong.

**Comment by AlCalzone** [comment]
Did you find out anything new? I double-checked and the only way `requestBase: 'coaps://undefined:5684/'` and an undefined hostname can slip its way into `TradfriClient` is if `undefined` was passed as the hostname to its constructor.

**Comment by AlCalzone** [comment]
In the meantime I changed the constructor to throw if an undefined hostname is passed. This should help determine if the problem arises before or after authentication.

**Comment by Boimb** [comment]
I'm on holidays for now. I'll check it out when I'll be back home.
Thx.

**Comment by AlCalzone** [comment]
I just ran your sample code on my raspberry, except with `gateway.name` instead of `gateway.host` and it works like a charm.

**Comment by Boimb** [comment]
The issue was due to the firmware and my router. Version 1.2.42 seems to encounter issues getting an ip with some routers (Livebox v4 from Orange in my case). I could solve this today when back at work, I connected the gateway to another router. 
Firmware updated, back home, everything ‘s ok. 
Thanks for your time trying to solve this issue that wasn’t cause by your module in any way. 
I close this. 
See ya.
